### PR TITLE
Extended core support by changing arguments interface for SET command

### LINKED ISSUE #992: Support new Redis 6.2 arguments for SET
Add support for the enw arguments in Redis 6.2 for SET as specified at https://redis.io/commands/SET.

Detail of update: Allowed the `NX` and `GET` options to be used together.

This is part of #820.
Note: Due to the autogenerated nature of this, some of this may be already implemented.

### PR DESCRIPTION
**Pull request by vladvildanov**
Closes #992

### CODE REVIEW

**Comment by tillkruss** [comment]
This would be a breaking change and should go into v3.x, eh?

**Comment by tillkruss** [comment]
> @tillkruss Looks like, there's a lot of command interface changes since Redis 6.2, so I assume it should be 3.x for all of them

Yes, we can't merge breaking changes into v2.

**Comment by chayim** [comment]
Folks, I don't think we're having the same conversation here. If an interface is extended, meaning existing user usage doesn't change - I'd argue we can continue. If we **break**  an interface, then yes.

@tillkruss This is similar to many other projects so WDYT? Again the proposal I'm making is that when extending the interface - meaning no user facing change or behavioural change unless actively used, we continue within this code path.

**Comment by tillkruss** [comment]
> Folks, I don't think we're having the same conversation here. If an interface is extended, meaning existing user usage doesn't change - I'd argue we can continue. If we **break** an interface, then yes.

Correct. I do see several changes to existing test cases. If we can leave the current ones in place and add new ones that prove the old functionality works, it's not a BC.

File: src/ClientContextInterface.php [hunk 1]
```diff
@@ -70,7 +70,7 @@
  * @method $this mset(array $dictionary)
  * @method $this msetnx(array $dictionary)
  * @method $this psetex($key, $milliseconds, $value)
- * @method $this set($key, $value, $expireResolution = null, $expireTTL = null, $flag = null)
+ * @method $this set($key, $value, bool $getOld = false, $expireResolution = null, $expireTTL = null, $flag = null)
```
Comment (chayim) [inline src/ClientContextInterface.php:73]: given the location. this looks breaking.
how about we punt this @vladvildanov

**Pull request closed without merging**